src/FeaturedComponents.jsx

```diff
@@ -1,78 +1,220 @@
-      src,
-    };
+const followGraph = context.accountId ? Social.keys(`${context.accountId}/graph/follow/*`, "final") : null;
+const accountsFollowing =
+  props.accountsFollowing ?? (followGraph ? Object.keys(followGraph[context.accountId].graph.follow || {}) : null);
 
-    components.push(result);
+const optionsMap = {
+    featured: "Featured",
+    followed: "From Followed",
+};
+
+function fetchGraphQL(operationsDoc, operationName, variables) {
+  return asyncFetch(`${GRAPHQL_ENDPOINT}/v1/graphql`, {
+    method: "POST",
+    headers: { "x-hasura-role": "kevin0_near" },
+    body: JSON.stringify({
+      query: operationsDoc,
+      variables: variables,
+      operationName: operationName,
+    }),
   });
 }
 
+const createQuery = () => {
+  console.log(accountsFollowing); //["roshaan.near", "calebjacob.near", "potlock.near"]
+  return `
+  query ComponentFollowingQuery($accountsFollowing: [String], $limit: Int) {
+    kevin0_near_component_01_info(
+      where: {component_author_id: {_in: $accountsFollowing}}
+      order_by: {block_height: desc}
+      limit: $limit,
+    ) {
+      component_author_id
+      fork_count
+      component_name
+      component_id
+      code
+      block_height
+      block_timestamp_ms
+      star_count
+      tags
+      image_ipfs_cid
+      description
+      website
+    }
+}
+    `;
+  };
+  
+const renderComponents = (sortOption) => {
+  setIsLoading(true);
+ 
+  const componentContainer = [];
+
+  if (sortOption === "featured") {
+    if (componentData) {
+      featuredComponentPaths.forEach((src) => {
+        const path = src.split("/");
+        const result = { metadata: componentData[path[0]][path[1]][path[2]].metadata, src,};
+        componentContainer.push(result);
+      });
+    }
+    setComponents(componentContainer);
+    setIsLoading(false);
+  } else if (sortOption === "followed") {
+  const queryName = "ComponentFollowingQuery";
+
+  fetchGraphQL(createQuery(), queryName, {
+    accountsFollowing: accountsFollowing,
+    limit: LIMIT,
+  } )
+  .then((result) => {
+      if (result.status === 200 && result.body) {
+        if (result.body.errors) { 
+          console.log("error:", result.body.errors);
+          return;
+        }
+        let data = result.body.data;
+        if (data) {
+          const components_item = data.kevin0_near_component_01_info;
+          
+          components_item.forEach((component) => {
+            const {
+              block_height,
+              block_timestamp_ms,
+              component_author_id,
+              component_id,
+              component_name,
+              fork_count,
+              star_count,
+              tags,
+              image_ipfs_cid,
+              description,
+              website,
+            } = component;
+            console.log('tags here', tags);
+            
+            if(tags){
+              tags.forEach((tag) => {
+
+              });
+            }
```

Comment:
Will want to remove the console log and update our logic here once we know the shape of `tags`. I believe it might be an array of strings.

Reply:
just a string separated by commas